src/openrct2/ride/VehicleRiderControl.cpp

```diff
@@ -0,0 +1,226 @@
+    { MPH(25), MPH(14), MPH(10), 6 },  { MPH(19), MPH(11), MPH(10), 8 },  { MPH(24), MPH(10), MPH(10), 8 },
+    { MPH(27), MPH(17), MPH(12), 7 },  { MPH(27), MPH(16), MPH(12), 5 },  { MPH(24), MPH(16), MPH(8), 7 },
+    { MPH(20), MPH(9), MPH(8), 9 },    { MPH(27), MPH(19), MPH(15), 4 },  { MPH(21), MPH(10), MPH(11), 8 },
+    { MPH(17), MPH(8), MPH(5), 11 },   { MPH(16), MPH(10), MPH(10), 8 },  { MPH(29), MPH(18), MPH(10), 7 },
+    { MPH(23), MPH(16), MPH(14), 5 },  { MPH(16), MPH(8), MPH(6), 10 },   { MPH(24), MPH(10), MPH(8), 11 },
+    { MPH(21), MPH(11), MPH(7), 11 },  { MPH(26), MPH(13), MPH(8), 7 },   { MPH(22), MPH(10), MPH(8), 11 },
+    { MPH(25), MPH(13), MPH(11), 9 },  { MPH(24), MPH(15), MPH(8), 5 },   { MPH(26), MPH(12), MPH(6), 8 },
+    { MPH(21), MPH(15), MPH(15), 7 },  { MPH(28), MPH(19), MPH(14), 6 },  { MPH(23), MPH(14), MPH(9), 8 },
+    { MPH(25), MPH(13), MPH(6), 9 },   { MPH(18), MPH(11), MPH(12), 7 },  { MPH(22), MPH(13), MPH(7), 6 },
+    { MPH(23), MPH(13), MPH(7), 8 },   { MPH(27), MPH(18), MPH(9), 5 },   { MPH(20), MPH(9), MPH(6), 10 },
+    { MPH(29), MPH(9), MPH(9), 12 },   { MPH(26), MPH(12), MPH(9), 11 },  { MPH(27), MPH(10), MPH(5), 11 },
+    { MPH(26), MPH(20), MPH(9), 3 },   { MPH(18), MPH(10), MPH(11), 8 },  { MPH(28), MPH(16), MPH(12), 8 },
+    { MPH(13), MPH(9), MPH(7), 9 },    { MPH(24), MPH(15), MPH(9), 6 },   { MPH(20), MPH(11), MPH(9), 8 },
+    { MPH(24), MPH(15), MPH(12), 4 },  { MPH(24), MPH(14), MPH(9), 5 },   { MPH(22), MPH(11), MPH(8), 10 },
+    { MPH(24), MPH(11), MPH(10), 10 }, { MPH(24), MPH(17), MPH(10), 7 },  { MPH(28), MPH(18), MPH(13), 7 },
+    { MPH(23), MPH(11), MPH(8), 12 },  { MPH(25), MPH(12), MPH(11), 8 },  { MPH(21), MPH(10), MPH(11), 8 },
+    { MPH(15), MPH(8), MPH(7), 10 },   { MPH(26), MPH(16), MPH(9), 8 },   { MPH(21), MPH(10), MPH(9), 10 },
+    { MPH(27), MPH(17), MPH(16), 6 },  { MPH(22), MPH(12), MPH(6), 9 },   { MPH(25), MPH(9), MPH(4), 11 },
+    { MPH(26), MPH(16), MPH(13), 9 },  { MPH(26), MPH(19), MPH(11), 6 },  { MPH(24), MPH(15), MPH(13), 7 },
+    { MPH(16), MPH(9), MPH(10), 8 },   { MPH(21), MPH(11), MPH(7), 6 },   { MPH(28), MPH(20), MPH(15), 6 },
+    { MPH(25), MPH(15), MPH(9), 6 }
+};
+
+int32_t Vehicle::CalculateRiderBraking() const
+{
+    if (num_peeps == 0)
+        return 0;
+
+    RiderControlSettings riderSettings = riderTable[peep[0].ToUnderlying() & 0xFF];
+
+    // Acceleration to apply when rider is braking
+    int32_t minBrake = (1 << 16);
+
+    // Acceleration to apply when rider is braking hard (used when going much faster than the rider's preferred speed or very
+    // close to the rider in front)
+    int32_t maxBrake = (12 << 16);
+
+    // Velocity above which riders will attempt to maintain separation from the vehicle in front
+    int32_t minFollowVelocity = MPH(4);
+
+    // Minimum separation distance that riders will allow (regardless of followDistance)
+    int32_t minFollowDistance = 32;
```

Comment:
Since these are not updated on the function and are based on literals, they should probably be `constexpr` variables defined at the top of the file.